dojo/tools/checkmarx_cxflow_sast/parser.py

```diff
@@ -0,0 +1,153 @@
+import json
+import dateutil.parser
+import logging
+
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+
+class _PathNode(object):
+    def __init__(self, file: str, line: str, column: str, node_object: str, length: str, snippet: str):
+        self.file = file
+        self.line = line
+        self.column = int(column)
+        self.node_object = node_object
+        self.length = int(length)
+        self.snippet = snippet
+
+    @classmethod
+    def from_json_object(cls, data):
+        return _PathNode(
+            data.get("file"),
+            data.get("line"),
+            data.get("column"),
+            data.get("object"),
+            data.get("length"),
+            data.get("snippet")
+        )
+
+
+class _Path(object):
+    def __init__(self, sink: _PathNode, source: _PathNode, state: str, paths: [_PathNode]):
+        self.sink = sink
+        self.source = source
+        self.state = state
+        self.paths = paths
+
+
+class CheckmarxCXFlowSastParser(object):
+    def __init__(self):
+        pass
+
+    def get_scan_types(self):
+        return ["Checkmarx CxFlow SAST"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Detailed Report. Import all vulnerabilities from checkmarx without aggregation"
+
+    def get_findings(self, file, test):
+        if file.name.strip().lower().endswith(".json"):
+            return self._get_findings_json(file, test)
+        else:
+            logger.warning(f"Not supported file format ${file}")
+            return list()
+
+    def _get_findings_json(self, file, test):
+        data = json.load(file)
+        findings = []
+        # deepLink = data.get("deepLink")
```

Comment:
If you don't need this line, maybe remove it?

Reply:
Yes